### DISCUSSION TITLE
Note taking templates and how to customize

### DISCUSSION THREAD

**Discussion by Kahros** [xournalpp/xournalpp#5065] [Q&A]
I just recently downloaded this application and loving it except for minor features of life.  Most notably, setting custom page templates/layouts.

I've figured this out so far by.
>Create the page layout I want,
>Export as PDF
>Open said PDF
>Set Configure Page Template
>Set new page pages to open with PDF background
>Select the PDF background from the options shown. 

This is a whole lot of work just to even start to write out notes.  I like to use multiple different page layouts, such as Double Journaling, Cornell, Calendar and Events, etc.

Is there an easier way to have this feature applied?  Am I doing something wrong/the long way??  Is there a method for having PDFs load/cache at launch, in an attempt to greatly reduce the amount of steps (ergo time) to load a new layout?

**Comment by rolandlo** [comment]
I'm not really using templates from PDF backgrounds myself, but I still have an idea which might be useful. Start by combining all layouts (Double Journaling, Cornell, Calendar, Events etc.) in a single PDF and create a `.xopp`-file from it where you delete all pages except the one with the layout you use the most. Then you can always start by loading that `.xopp`-file and then either just start to copy that document background, or first replace the pdf page used for that page.

Improving even further you could write a [plugin](https://xournalpp.github.io/guide/plugins/plugins/), where you use
`app.changeBackgroundPdfPageNr` to change the background of the current page to one of your layouts.
In case you already have multiple pages you would also need  `app.getDocumentStructure` to get the number of pages and `app.setCurrentPage` to iterate through the pages to change the background of all pages to one of your layouts.
Then all it takes is pressing a keyboard shortcut or using the plugins menu to switch to one of your layout.

It may sound a bit complicated (and yes, we should have a simpler solution for that), but it's basically a one time effort and I could help you writing the plugin if you are interested.

**Reply by Kahros** [reply]
I was also thinking on a plugin, but I'm not good at programming (just started Basic Programming class for college).  Even having a plugin that prefetches the PDF at launch, so it's selectable as a PDF background(s) would be nice.  

I'm having other issues with the program, mainly that my pen strokes are heavily delayed at starting strokes when zoomed in on a page, and creature comforts - like having a list of books made on a startup screen vs relying on the file explorer.

The latter feels like it can be done as plugin.

**Reply by Kahros** [reply]
Update, I resolved my beginning stroke lag by disabling the windows Wacom service "WTabletServicesISD".  Created an issue where inertia scrolling was freaking out.  Disabling it fixed it.
